File: backend/src/test/java/org/luxons/sevenwonders/test/api/ApiPlayer.java
package org.luxons.sevenwonders.test.api;

import java.util.Objects;

public class ApiPlayer {

    private String username;

    private String displayName;

    private int index;

    public String getUsername() {
        return username;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public int getIndex() {
        return index;
    }

    public void setIndex(int index) {
        this.index = index;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        ApiPlayer apiPlayer = (ApiPlayer) o;
        return index == apiPlayer.index && Objects.equals(username, apiPlayer.username) && Objects.equals(displayName,
                apiPlayer.displayName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(username, displayName, index);
    }
}

File: backend/src/test/kotlin/org/luxons/sevenwonders/test/api/ApiPlayer.kt
package org.luxons.sevenwonders.test.api

import java.util.Objects

class ApiPlayer {

    val username: String? = null

    var displayName: String? = null

    var index: Int = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val apiPlayer = other as ApiPlayer?
        return index == apiPlayer!!.index && username == apiPlayer.username && displayName == apiPlayer.displayName
    }

    override fun hashCode(): Int {
        return Objects.hash(username, displayName, index)
    }
}
